frontend/js/src/utils/utils.tsx

```diff
@@ -131,6 +131,61 @@ const searchForYoutubeTrack = async (
   return null;
 };
 
+const searchForSoundcloudTrack = async (
+  soundcloudToken?: string,
+  trackName?: string,
+  artistName?: string,
+  releaseName?: string
+): Promise<string | null> => {
+  if (!soundcloudToken) return null;
+  let query = trackName ?? "";
+  if (artistName) {
+    query += ` ${artistName}`;
+  }
+  // Considering we cannot tell the Soundcloud API that this should match only an album title,
+  // results are paradoxically sometimes worse if we add it to the query
+  if (releaseName) {
+    query += ` ${releaseName}`;
+  }
+  if (!query) {
+    return null;
+  }
+
+  const response = await fetch(
+    `https://api.soundcloud.com/tracks?q=${encodeURIComponent(
+      query
+    )}&access=playable`,
+    {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `OAuth ${soundcloudToken}`,
+      },
+    }
+  );
+  const responseBody = await response.json();
+  if (!response.ok) {
+    throw responseBody;
+  }
+  const streamUrl = responseBody?.[0]?.stream_url;
+  if (!streamUrl) {
+    return null;
+  }
+
+  const response2 = await fetch(streamUrl, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `OAuth ${soundcloudToken}`,
+    },
+  });
+  const responseBody2 = await response2.json();
+  if (!response.ok) {
+    throw responseBody;
+  }
+  return responseBody2?.http_mp3_128_url;
```

Comment:
Just checking: is this prop fully stable in the response?
The live API docs do seem to say so.
In any case, this code would just safely return undefined otherwise.

Reply:
Yup, this one should be stable.